src/components/vuestic-components/va-infinite-scroll/VaInfiniteScroll.demo.vue

```diff
@@ -0,0 +1,216 @@
+              v-for="(record, index) in records"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+
+      <VbCard title="selector target">
+        <div
+          id="target"
+          class="scroll__container"
+        >
+          <va-infinite-scroll
+            scroll-target="#target"
+            :load="appendRecordsAsync"
+          >
+            <li
+              v-for="(record, index) in records"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+
+      <VbCard title="debounce">
+        <div
+          class="scroll__container"
+        >
+          <va-infinite-scroll
+            :debounce="5000"
+            :load="appendRecordsAsync"
+          >
+            <li
+              v-for="(record, index) in records"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+
+      <VbCard title="load failed">
+        <div
+          class="scroll__container"
+        >
+          <va-infinite-scroll
+            :load="failedLoadAsync"
+          >
+            <li
+              v-for="(record, index) in records"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+
+      <VbCard title="reverse">
+        <div class="scroll__container">
+          <va-infinite-scroll
+            reverse
+            :load="appendRecordsAsync"
+          >
+            <li
+              v-for="(record, index) in records"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+
+      <VbCard title="disabled">
+        <div class="scroll__container">
+          <va-infinite-scroll
+            disabled
+            :load="appendRecordsAsync"
+          >
+            <li
+              v-for="(record, index) in disabledRecords"
+              :key="record.id"
+            >
+              {{ record.text }} #{{ index }}
+            </li>
+          </va-infinite-scroll>
+        </div>
+      </VbCard>
+    </VbDemo>
+  </div>
+</template>
+
+<script>
+import VaInfiniteScroll from './VaInfiniteScroll'
+import VaProgressCircle from '../va-progress-bar/progress-types/VaProgressCircle'
+import { times } from 'lodash'
+import { sleep } from '../../../services/utils'
+
+export default {
+  components: {
+    VaInfiniteScroll,
+    VaProgressCircle,
+  },
+  data () {
+    return {
+      records: this.getInitialRecords(),
+      disabledRecords: this.getInitialRecords(),
+    }
+  },
+  methods: {
+    getInitialRecords () {
```

Comment:
You can declare these methods outside of component, given these are just factories and do not rely on `this`.